mvlearn/embed/kcca.py

```diff
@@ -201,6 +210,12 @@ def __init__(
         if self.constant < 0 or not (type(self.constant) == float
                                      or type(self.constant) == int):
             raise ValueError("constant must be a positive integer")
+        if self.decomp == "icd":
+            if self.mrank < 0 or not (type(self.mrank) == int):
+                raise ValueError("mrank must be a positive integer no greater \
+                                 than the number of components")
```

Comment:
Does this also need to check if mrank is no greater than n_components?

Reply:
yeah, i had it and then removed when bug fixing and meant to add it back. adding it back now